### PR TITLE
Add delete dashboard button

### PR DESCRIPTION
**Pull request by john-dupuy**
* Adds a delete button for removing a dashboard (I just picked a "-" icon, let me know if you'd prefer a different style @rsnyman)
* Adds a delete dash modal
* On delete the user is taken to the empty dashboard state

Screenshot from 2020-12-08 15-48-13

Screenshot from 2020-12-08 15-48-50

### CODE REVIEW

File: frontend/src/dashboard.js [hunk 1]
```diff
@@ -246,6 +271,17 @@ export class Dashboard extends React.Component {
                   <PlusCircleIcon />
                 </Button>
               </FlexItem>
+              <FlexItem>
```
Comment (rsnyman) [inline frontend/src/dashboard.js:274]: Add ` spacer={{ default: 'spacerNone' }}` to the PREVIOUS `FlexItem` to get rid of the huge space between these two.

File: frontend/src/dashboard.js [hunk 2]
```diff
@@ -19,11 +19,11 @@ import {
   Text,
   Title,
 } from '@patternfly/react-core';
-import { ArchiveIcon, CubesIcon, PlusCircleIcon, TachometerAltIcon } from '@patternfly/react-icons';
+import { ArchiveIcon, CubesIcon, MinusCircleIcon, PlusCircleIcon, TachometerAltIcon } from '@patternfly/react-icons';
```
Comment (rsnyman) [inline frontend/src/dashboard.js:22]: Use `TimesCircleIcon` instead.

File: frontend/src/components/delete-dashboard-modal.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        variant={ModalVariant.small}
+        title="Delete dashboard"
+        isOpen={this.props.isOpen}
+        onClose={this.onClose}
+        actions={[
+          <Button key="delete" variant="danger" onClick={this.onDelete}>Delete</Button>,
+          <Button key="cancel" variant="link" onClick={this.onClose}>Cancel</Button>
+        ]}
+      >
+      <Text>Would you like to delete the current dashboard? All widgets on the dashboard will also be deleted.</Text>
```
Comment (rsnyman) [inline frontend/src/components/delete-dashboard-modal.js:40]: I think we need to emphasise "all widgets" more... something like:

```
<strong>ALL WIDGETS</strong>
```

**Review by rsnyman — CHANGES_REQUESTED** [review]

**Comment by john-dupuy** [comment]
OK pushed those changes and updated the screenshots to reflect them.

**Review by rsnyman — APPROVED** [review]

**Pull request merged**